src/components/Offers/Edit/EditOfferForm.js

```diff
@@ -0,0 +1,169 @@
+                message: "You are not authorized to edit this offer",
+            },
+        },
+    };
+
+    const {
+        reset,
+        submit,
+        setLoading,
+        setRequestErrors,
+        setOfferId,
+        setSuccess,
+        ...offerFormParams
+    } = useOfferForm(EditOfferSchema);
+
+    useEffect(() => {
+        if (offer && !isValidating && canEdit) {
+            reset(parseOfferForm(offer));
+        }
+    }, [canEdit, isValidating, offer, reset]);
+
+    const handleSubmit = useCallback(
+        (data) => {
+            setLoading(true);
+            const [jobMinDuration, jobMaxDuration] = data.jobDuration;
+            const publishDateChanged = data.publishDate.getTime() !== new Date(offer?.publishDate).getTime();
+            const publishEndDateChanged = data.publishEndDate.getTime() !== new Date(offer?.publishEndDate).getTime();
+            editOffer({
+                offerId: id,
+                ...data,
+                vacancies: data.vacancies || undefined,
+                publishDate: publishDateChanged ? data.publishDate : undefined,
+                publishEndDate: publishEndDateChanged ? data.publishEndDate : undefined,
+                contacts: data.contacts.map((val) => val.value),
+                requirements: data.requirements.map((val) => (val, val.value)),
+                isPaid: data.isPaid === "none" ? undefined : data.isPaid,
+                jobStartDate: !data.jobStartDate ? undefined : data.jobStartDate,
+                jobMinDuration,
+                jobMaxDuration,
+            })
+                .then((obj) => {
+                    setRequestErrors({});
+                    setOfferId(obj._id);
+                    setLoading(false);
+                    setSuccess(true);
+                })
+                .catch((err) => {
+                    const reqErrors = parseRequestErrors(err);
+                    setRequestErrors(reqErrors);
+                    setLoading(false);
+                });
+        },
+        [id, offer, setLoading, setOfferId, setRequestErrors, setSuccess],
+    );
+
+    return {
+        controllerOptions: {
+            initialValue: {
+                ...offerFormParams,
+                submit: submit(handleSubmit),
+                reset,
+                setLoading,
+                setRequestErrors,
+                setSuccess,
+                offerId: id,
+                errorOffer,
+                loadingOffer,
+                redirectProps,
+                user,
+                isValidating,
+                canEdit,
+            },
+        },
+    };
+};
+
+const EditOfferForm = () => {
+    const {
+        loadingOffer,
+        errorOffer,
+        redirectProps,
+        isValidating,
+        canEdit,
+    } = useContext(EditOfferControllerContext);
+
+    if (errorOffer || (!loadingOffer && !isValidating && !canEdit)) {
+
+        return (
+            <Redirect {...redirectProps} />
+        );
+    }
```

Comment:
This is so simple that we can make it smaller, to be easier to read
```suggestion
    if (errorOffer || (!loadingOffer && !isValidating && !canEdit))
        return <Redirect {...redirectProps} />
    
```